core/vm/contracts.go

```diff
@@ -294,16 +242,14 @@ func ActivePrecompiles(rules params.Rules) []common.Address {
 // - the returned bytes,
 // - the _remaining_ gas,
 // - any error that occurred
-func RunPrecompiledContract(p PrecompiledContract, input []byte, suppliedGas uint64, caller common.Address, evm *EVM) (ret []byte, remainingGas uint64, err error) {
-	log.Trace("Running precompiled contract", "input", input, "gas", suppliedGas)
+func RunPrecompiledContract(p CeloPrecompiledContract, input []byte, suppliedGas uint64, ctx *celoPrecompileContext) (ret []byte, remainingGas uint64, err error) {
 	gasCost := p.RequiredGas(input)
 	if suppliedGas < gasCost {
 		return nil, 0, ErrOutOfGas
 	}
-	gasLeft := suppliedGas - gasCost
-	output, err := p.Run(input, caller, evm)
-	log.Trace("Finished running precompiled contract", "input", input, "gas", suppliedGas, "gas_left", gasLeft)
-	return output, gasLeft, err
+	suppliedGas -= gasCost
+	output, err := p.Run(input, ctx)
+	return output, suppliedGas, err
```

Comment:
The naming `suppliedGas` is not the best fit after the gasCost is deducted. I prefer the old version with the `gasLeft`. Or just return `output, suppliedGas - gasCost, err` if you don't like the additional local variable.

Reply:
Makes sense, I changed it back.